src/pyaro_readers/eeareader/EEATimeseriesReader.py

```diff
@@ -1,300 +1,484 @@
 
-                # Filters out invalid data
-                lf = lf.filter(polars.col(PARQUET_FIELDS["flags"]) > 0)
+    @property
+    def altitudes(self) -> np.ndarray:
+        return np.array(self._data["Altitude"], dtype=float)
 
-                # Changes timezones
-                lf = lf.with_columns(
-                    polars.col(PARQUET_FIELDS["start_times"])
-                    .dt.replace_time_zone("Etc/GMT-1")
-                    .dt.convert_time_zone("UTC")
-                    .alias(PARQUET_FIELDS["start_times"])
-                )
+    @property
+    def start_times(self) -> np.ndarray:
+        return np.array(self._data["Start"])
 
-                lf = lf.with_columns(
-                    polars.col(PARQUET_FIELDS["end_times"])
-                    .dt.replace_time_zone("Etc/GMT-1")
-                    .dt.convert_time_zone("UTC")
-                    .alias(PARQUET_FIELDS["end_times"])
-                )
+    @property
+    def end_times(self) -> np.ndarray:
+        return np.array(self._data["End"])
 
-                file_datapoints = lf.select(polars.len())[0, 0]
+    @property
+    def flags(self) -> np.ndarray:
+        def mapper(value: int) -> int:
+            if value == 1:
+                return pyaro.timeseries.Flag.VALID
+            elif value == 2 or value == 3:
+                return pyaro.timeseries.Flag.BELOW_THRESHOLD
+            else:
+                return pyaro.timeseries.Flag.INVALID
+
+        valid = self._data["Validity"].map_elements(mapper, return_dtype=int)
+        return np.array(valid)
+
+    @property
+    def standard_deviations(self) -> np.ndarray:
+        return np.repeat(np.nan, self._nrecords())
+
+    def _nrecords(self) -> int:
+        return self._data.shape[0]
+
+
+def _read(filepath: Path, pyarrow_filters) -> polars.DataFrame:
+    # TODO: Timezone fixup??
+    return polars.read_parquet(
+        filepath,
+        use_pyarrow=True,
+        pyarrow_options={"filters": pyarrow_filters},
+        columns=[
+            "Samplingpoint",
+            "Pollutant",
+            "Start",
+            "End",
+            "Value",
+            "Unit",
+            "Validity",
+        ],
+    ).cast({"Value": polars.Float32})
+
+
+@dataclasses.dataclass
+class _Filters:
+    pyarrow: list[tuple[str, str, datetime]]
+    country: pyaro.timeseries.Filter.CountryFilter | None
+    time: pyaro.timeseries.Filter.TimeBoundsFilter | None
+
+
+def _pyarrow_timefilter(
+    filter: pyaro.timeseries.Filter.TimeBoundsFilter,
+) -> list[tuple[str, str, datetime]]:
+    # Time filtering might not be expressible as pyarrow filters alone,
+    # so we supply a coarse filter which should be filtered later on
+    # TODO: Make this support more filtering whilst reading
+    min_time, max_time = filter.envelope()
+
+    return [
+        ("Start", ">=", min_time),
+        ("Start", "<=", max_time),
+        ("End", ">=", min_time),
+        ("End", "<=", max_time),
+    ]
```

Comment:
Which version of `polars` and `pyarrow` this works with?
When I implemented this kind of filters for the CAMS2_35b API I had to use a `pyarrow.scalar(datetime)` for the filter bounds, and ended up with something like this:
``` python
from datetime import datetime

import polars as pl
import pyarrow as pa

[...]

date_ranges: list[tuple(datetime, datetime)]
filters = [
    ## pyarrow rejected this filter
    # [("time", ">=", dates[0]), ("time", "<=", dates[1])]
    ## this worked
    [("time", ">=", pa.scalar(dates[0])), ("time", "<=", pa.scalar(dates[1]))]   
    for dates in date_ranges
]

df = pl.read_parquet(
        path,
        columns=["station", "time", classification_column],
        use_pyarrow=True,
        pyarrow_options=dict(filters=filters),
)
```

Reply:
```
pyarrow==18.0.0
polars==1.12.0
```

Reply:
Thanks, this will simplify the CAMS2_35b API code